internal/bus/message_pipe.go

```diff
@@ -21,14 +21,19 @@ type (
 		Data  Payload
 	}
 
+	MessageWithContext struct {
+		ctx     context.Context
+		message *Message
+	}
```

Comment:
Have we considered adding `context.Context` inside the `Message` struct?

Reply:
The reason for the `MessageWithContext` is that we dont want to be passing around the context inside a struct in the plugins. Its only used for sending the context and message over the message pipe channel.

I don't think it would be good to change the plugin Process function from `Process(ctx context.Context, msg *Message)` to `Process(msg *Message)` as its not clear that the message has the context. Always best to have the context as the first parameter in a function.